ext/native/thin3d/VulkanRenderManager.cpp

```diff
@@ -180,26 +190,45 @@ void VulkanRenderManager::CreateBackbuffers() {
 	// Start the thread.
 	if (useThread) {
 		run_ = true;
+		// Won't necessarily be 0.
+		threadInitFrame_ = vulkan_->GetCurFrame();
 		thread_ = std::thread(&VulkanRenderManager::ThreadFunc, this);
 	}
 }
 
-void VulkanRenderManager::DestroyBackbuffers() {
-	if (useThread) {
+void VulkanRenderManager::StopThread(bool shutdown) {
+	if (useThread && run_) {
 		run_ = false;
 		// Stop the thread.
 		for (int i = 0; i < vulkan_->GetInflightFrames(); i++) {
+			auto &frameData = frameData_[i];
 			{
-				std::unique_lock<std::mutex> lock(frameData_[i].push_mutex);
-				frameData_[i].push_condVar.notify_all();
+				std::unique_lock<std::mutex> lock(frameData.push_mutex);
+				frameData.push_condVar.notify_all();
 			}
 			{
-				std::unique_lock<std::mutex> lock(frameData_[i].pull_mutex);
-				frameData_[i].pull_condVar.notify_all();
+				std::unique_lock<std::mutex> lock(frameData.pull_mutex);
+				frameData.pull_condVar.notify_all();
 			}
 		}
 		thread_.join();
+
+		// Reset fences for next time around - must be done after join.
+		for (int i = 0; i < vulkan_->GetInflightFrames(); i++) {
+			auto &frameData = frameData_[i];
+			frameData.readyForRun = false;
+			if (!shutdown && !frameData.readyForFence) {
+				vkDestroyFence(vulkan_->GetDevice(), frameData.fence, nullptr);
```

Comment:
What about vkResetFences? Unless we zap the whole device, fences should be reusable.

(Hm, I made this comment before but it got lost?)

Reply:
Oops, yeah.  Typed that when I had only moved the flags so far...

-[Unknown]